helix-core/src/main/java/org/apache/helix/manager/zk/ZKUtil.java

```diff
@@ -632,4 +634,39 @@ private static RealmAwareZkClient getHelixZkClient(String zkAddr) {
     return DedicatedZkClientFactory.getInstance()
         .buildZkClient(new HelixZkClient.ZkConnectionConfig(zkAddr), clientConfig);
   }
+
+  /**
+   * Convert Stat fields into a Map.
+   *     private long czxid;
+   *     private long mzxid;
+   *     private long ctime;
+   *     private long mtime;
+   *     private int version;
+   *     private int cversion;
+   *     private int aversion;
+   *     private long ephemeralOwner;
+   *     private int dataLength;
+   *     private int numChildren;
+   *     private long pzxid;
+   * @param stat
+   * @return
+   */
+  public static Map<String, String> fromStatToMap(Stat stat) {
```

Comment:
You might want to checkout this method of the Zookeeper Stat class, simply implement an OutputArchive to avoid hardcode getting fields here.

  public void serialize(OutputArchive a_, String tag) throws java.io.IOException {
    a_.startRecord(this,tag);
    a_.writeLong(czxid,"czxid");
    a_.writeLong(mzxid,"mzxid");
    a_.writeLong(ctime,"ctime");
    a_.writeLong(mtime,"mtime");
    a_.writeInt(version,"version");
    a_.writeInt(cversion,"cversion");
    a_.writeInt(aversion,"aversion");
    a_.writeLong(ephemeralOwner,"ephemeralOwner");
    a_.writeInt(dataLength,"dataLength");
    a_.writeInt(numChildren,"numChildren");
    a_.writeLong(pzxid,"pzxid");
    a_.endRecord(this,tag);
  }

Reply:
@jiajunwang I already did consider this option, but OutputArchive/serialize is used for different purposes. This requires more code and adds complexity/dependency on apache.jute which we don't simply need.

Reply:
Agreed. Great that you checked that option already.